src/embedding/embedding-service.ts

```
@@ -0,0 +1,147 @@
+            config.embeddingModel,
+            config.dimensions
+          );
+
+import { parse } from 'shell-quote';
+// ... existing imports ...
+        case 'local-cli':
+          if (!config.cliCommand) {
+            console.warn('CLI command not provided. Embedding generation will be disabled.');
+            return null;
+          }
+          const commandParts = parse(config.cliCommand) as string[];
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)